Sources/NIO/Codec.swift

```diff
@@ -222,13 +222,13 @@ extension ByteToMessageDecoder {
     public func shouldReclaimBytes(buffer: ByteBuffer) -> Bool {
         // We want to reclaim in the following cases:
         //
-        // 1. If there is more than 2kB of memory to reclaim
+        // 1. If there is at least 2kB of memory to reclaim
         // 2. If the buffer is more than 50% reclaimable memory and is at least
         //    1kB in size.
-        if buffer.readerIndex > 2048 {
+        if buffer.readerIndex >= 2048 {
             return true
         }
-        return buffer.capacity > 1024 && (buffer.capacity - buffer.readerIndex) >= buffer.readerIndex
+        return buffer.capacity >= 1024 && (buffer.capacity - buffer.readerIndex) < buffer.readerIndex
```

Comment:
- the first change `>` --> `>=` essentially halves the threshold. We only allocate ByteBuffers in powers of 2. So outside of slices, we should only ever see a `1024` capacity or a `2048` capacity BB, nothing in between.
- I'd revert this change and go back to `>` because it's unrelated.
- let's use `.storageCapacity` even though in this case it's debatable I think

Reply:
Fixed.

Reply:
thank you